data/data/openstack/topology/private-network.tf

```diff
@@ -66,8 +66,9 @@ resource "openstack_networking_port_v2" "lb_port" {
 }
 
 data "openstack_networking_network_v2" "external_network" {
-  name     = "${var.external_network}"
-  external = true
+  name       = "${var.external_network}"
+  network_id = "${var.external_network_id}"
```

Comment:
What happens if someone "accidentally" sets both, name and id, and these 2 don't correspond to the same network? Will neutron fail or will it give precedence to the id?

Reply:
I did test this.  Terraform will fail because it can't find a matching network with that name + UUID.